### DISCUSSION TITLE
Using a prefix in tabs and windows name

### DISCUSSION THREAD

**Discussion by HacksPloiter** [nvim-lualine/lualine.nvim#1163] [Configuration]
Edit: Correct title: Using a prefix in tabs and windows name

Hi @shadmansaleh, @acksid, @fitrh,

Could you please help me with below?
When I use component "tabs", I want the tabs name to come up with below naming pattern:
Tab 1 > Tab 2 > Tab 3 and so on.

In the same manner when I use the component "windows" with mode = 1, I want the windows name to come up with below pattern:
Window 1 > Window 2 > Window 3 and so on.

Can you please guide me on how to achieve above?
Thanks.

**Comment by fitrh** [comment]
I believe you can use the `fmt` options

```lua
{
  'tabs',
  fmt = function (content, context)
    return 'Tab ' .. content
  end,
}
```

And I think you can also apply it to window component.
Sorry I couldn't test it since I don't use lualine anymore

**Reply by HacksPloiter** [reply]
Thanks for the reply @fitrh.

I was able to achieve this for "tab" using
```
   93       lualine_a = {
   94         {
   95           'tabs',
   96           mode = 1,
   97           fmt = function(name,context)
   98             return 'Tab '..context.tabnr
   99           end
  100         },
  101       },
```
However, it is returning number from 1000 for "windows" when I use below:
```
  111           'windows',
  112           mode=2,
  113           icons_enabled = false,
  114           fmt = function(name, context)
  115             return 'Window '..context.winnr
  116           end,
  117         },
```

Please let me know know if you have any clue on this.
Thanks.

**Reply by HacksPloiter** [reply]
Also, may I please ask which plugin are you using now @fitrh ? And how is it different from Lualine?

**Reply by fitrh** [reply]
> However, it is returning number from 1000 for "windows"

`winnr` gives you a 'Window Handle', I think what you want is `win_number`


> Also, may I please ask which plugin are you using now @fitrh ? And how is it different from Lualine?

I wrote my own statusline, so it is simpler (it's a single file) than lualine

**Reply by HacksPloiter** [reply]
Thank you very much @fitrh.
It worked and was really helpful😊!

For people who want to do the same:
```
  110         {
  111           'windows',
  112           mode=0,
  113           icons_enabled = false,
  114           fmt = function(name, context)
  115             return 'Window '..context.win_number
  116           end,
  117         },
```
